packages/host/app/components/pill.gts

```diff
@@ -5,85 +5,105 @@ import Component from '@glimmer/component';
 
 import {
   IconButton,
-  Pill,
+  Pill as BoxelPill,
   RealmIcon,
   Switch,
 } from '@cardstack/boxel-ui/components';
-import { cn } from '@cardstack/boxel-ui/helpers';
-import { IconX } from '@cardstack/boxel-ui/icons';
+import { cn, cssVar } from '@cardstack/boxel-ui/helpers';
+import { IconX, CodeFile } from '@cardstack/boxel-ui/icons';
+
+import { isCardInstance } from '@cardstack/runtime-common';
 
 import { type CardDef } from 'https://cardstack.com/base/card-api';
 
+import { type FileDef } from 'https://cardstack.com/base/file-api';
+
 import RealmService from '../services/realm';
 
-interface CardPillSignature {
+interface PillSignature {
   Element: HTMLDivElement | HTMLButtonElement;
   Args: {
-    card: CardDef;
-    isAutoAttachedCard?: boolean;
-    removeCard?: (card: CardDef) => void;
+    item: CardDef | FileDef;
+    isAutoAttached?: boolean;
+    remove?: (item: CardDef | FileDef) => void;
     onToggle?: () => void;
     isEnabled?: boolean;
   };
 }
 
-export default class CardPill extends Component<CardPillSignature> {
+export default class Pill extends Component<PillSignature> {
   @service declare realm: RealmService;
 
   get component() {
-    return this.args.card.constructor.getComponent(this.args.card);
+    return this.args.item.constructor.getComponent(this.args.item);
   }
 
   get hideIconRight() {
-    return !this.args.onToggle && !this.args.removeCard;
+    return !this.args.onToggle && !this.args.remove;
+  }
+
+  get id() {
```

Comment:
There is an alternative approach in https://github.com/cardstack/boxel/pull/2135 where we have a `CardPill` and a `FilePill`. Perhaps that is better than having so many conditions in a `Pill`.

Reply:
Let me know your thoughts, we'll have to consolidate this otherwise there will be a massive conflict

Reply:
@jurgenwerk I rebased this PR to your branch.